test/extended/networking/services.go

```diff
@@ -0,0 +1,79 @@
+package networking
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/test/e2e"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("[networking] services", func() {
+	f1 := e2e.NewFramework("net-isolation1")
+	f2 := e2e.NewFramework("net-isolation2")
+
+	It("should allow connections to another pod on the same node via a service IP", func() {
+		Expect(checkService(f1, f1, false)).To(Succeed())
+	})
+
+	It("should allow connections to another pod on a different node via a service IP", func() {
+		Expect(checkService(f1, f1, true)).To(Succeed())
+	})
+
+	It("should allow connections to another pod in a different namespace on the same node via a service IP", func() {
+		if pluginIsolatesNamespaces() {
+			e2e.Skipf("Network plugin implements isolation.")
+		} else {
+			Expect(checkService(f1, f2, false)).To(Succeed())
+		}
+	})
+
+	It("should allow connections to another pod in a different namespace on a different node via a service IP", func() {
+		if pluginIsolatesNamespaces() {
+			e2e.Skipf("Network plugin implements isolation.")
+		} else {
+			Expect(checkService(f1, f2, true)).To(Succeed())
+		}
+	})
+
+	It("should prevent connections to another pod in a different namespace on the same node via a service IP", func() {
+		if pluginIsolatesNamespaces() {
+			err := checkService(f1, f2, false)
+			Expect(err).To(HaveOccurred())
+		} else {
+			e2e.Skipf("Network plugin does not implement isolation.")
+		}
+	})
+
+	It("should prevent connections to another pod in a different namespace on a different node via a service IP", func() {
+		if pluginIsolatesNamespaces() {
+			err := checkService(f1, f2, true)
+			Expect(err).To(HaveOccurred())
+		} else {
+			e2e.Skipf("Network plugin does not implement isolation.")
+		}
+	})
+})
+
+func checkService(serverFramework, clientFramework *e2e.Framework, differentNodes bool) error {
```

Comment:
The function is used for both isolated and non-isolated (basic sanity-checking) service testing